builtin/common/vector.lua

```diff
@@ -1,6 +1,18 @@
 
 vector = {}
 
+function vector.from(x, y, z)
+	return {x = x, y = y, z = z}
+end
+
+function vector.zero()
+	return {x = 0, y = 0, z = 0}
+end
+
+function vector.copy(v)
+	return {x = v.x, y = v.y, z = v.z}
+end
+
 function vector.new(a, b, c)
```

Comment:
`vector.new()` could be changed that way :

```
function vector.new(a, b, c)
	if type(a) == "table" then
		return vector.copy(a)
	elseif a then
		return vector.from(a, b, c)
	end
	return vector.zero()
end
```

This is a bit slower but I guess it'll be deprecated in (far) future.

Or, if merging `from` and `new`:

```
function vector.new(a, b, c)
	if a and b and c then
		return { x = a, y = b, z = c }
	end

	-- Future deprecated part
	if type(a) == "table" then
		return vector.copy(a)
	else
		return vector.zero()
	end
end
```

Reply:
Done.